### ISSUE TITLE
state changes after restart

### ISSUE THREAD

**Issue by p4co86** [philip1986/pimatic-led-light#38]
for example, devices are off, after a restart they are on. so laststate is not saved.

**Comment by michbeck100** [comment]
I think saving the last state is nice, because it could have changed in the meantime. I think it would be better to ask the led light for it's current state, when starting. Only if this is not possible, saving the state makes sense.

**Comment by mwittig** [comment]
@michbeck100 Generally, you're right. Note, however, some of the LED driver protocols are unidirectional! For Milight, for example, it is not possible to query the state.

**Comment by dandresm** [comment]
It would be nice to have the option to not trigger the switchs on pimatic start. The actual problem is that when pimatic or raspberry restart, all the lights gets switched on.

**Comment by michbeck100** [comment]
Actually the lights should just be switched on if the saved state says that they have been switched on, when pimatic was stopped. At least that's what this piece of code tries to do:

`
if @power then @turnOn() else @turnOff()
`

This code is in the constructor for most devices, but i think this doesn't work, because the default value of  `@power` is "off", and "off" is not a valid value for a boolean, so the expression will always call `turnOn()`.

**Comment by mwittig** [comment]
@michbeck100 You're right. this is a design flaw of the plugin. "true" and "false" shall be used.

**Comment by Xento** [comment]
Yeah it should be completly switched to true and false.
When this is done the pimatic-hap has to be switched to true and false, too.

**Comment by michbeck100** [comment]
https://github.com/philip1986/pimatic-led-light/pull/66 should fix this
